numpy/_core/src/multiarray/descriptor.c

```diff
@@ -2744,7 +2748,14 @@ arraydescr_reduce(PyArray_Descr *self, PyObject *NPY_UNUSED(args))
     }
     PyTuple_SET_ITEM(state, 5, PyLong_FromLong(elsize));
     PyTuple_SET_ITEM(state, 6, PyLong_FromLong(alignment));
-    PyTuple_SET_ITEM(state, 7, PyLong_FromLong(self->flags));
+    /* Old NumPy stored potentially negative values, so use char if possible */
```

Comment:
Are we doing this so that dtypes pickled in new numpy can still be read on old? My tendency would be to always write the new format, but be kind on reading the old one.

Reply:
Fair, removed it.